In [4]:
import pandas as pd

## Dataset 4

### Secção 1

explicar de forma detalhada o pré-processamento.

explicar porque nos metadados dos pacientes eliminamos as colunas e não as linhas

### Dados

In [5]:
data_ucec = pd.read_csv('/Users/adrianacosta/Documents/GitHub/Bioinformatica_23-24/Dataset4/data_RNA_Seq_v2_expression_median.txt', sep='\t')

In [7]:
data_ucec_hugo = data_ucec[data_ucec.notna()['Hugo_Symbol']]          
data_ucec_hugo = data_ucec_hugo.set_index("Hugo_Symbol")
del data_ucec_hugo["Entrez_Gene_Id"]

In [8]:
data_ucec_hugo.head()

TCGA-2E-A9G8-01  TCGA-4E-A92E-01  TCGA-5B-A90C-01  \
Hugo_Symbol                                                      
A1BG                 24.2393          19.8087           9.0662   
A1CF                  0.5157           0.0000           0.3163   
A2BP1                 0.0000           1.3661           1.5817   
A2LD1                73.8834         164.3720          75.2467   
A2M                6611.9100       13119.3000        3898.1000   

             TCGA-5S-A9Q8-01  TCGA-A5-A1OH-01  TCGA-A5-A2K2-01  \
Hugo_Symbol                                                      
A1BG                 60.9811         251.1470         128.9960   
A1CF                  0.0000           0.0000           0.0000   
A2BP1                 0.0000           2.2263           0.0000   
A2LD1                71.9793         239.0980          97.9912   
A2M                4464.3200        7509.7600        3453.6900   

             TCGA-A5-A2K3-01  TCGA-A5-A2K4-01  TCGA-A5-A2K5-01  \
Hugo_Symbol                                                      
A1BG                470.8330          66.7799          151.617   
A1CF                  0.9251           0.0000            0.000   
A2BP1                 0.0000           0.5394            0.000   
A2LD1               165.7820         109.0610          185.920   
A2M                8335.5600        7725.9900         5675.450   

             TCGA-A5-A2K7-01  ...  TCGA-FI-A2D2-01  TCGA-FI-A2D4-01  \
Hugo_Symbol                   ...                                     
A1BG                 99.4890  ...        33.660261        51.353706   
A1CF                  0.0000  ...         0.065503         0.065503   
A2BP1                 0.4667  ...        -0.330564        -0.330564   
A2LD1               154.0310  ...        57.648992       148.270803   
A2M                3424.5600  ...      6569.434216      4646.023218   

             TCGA-FI-A2D5-01  TCGA-FI-A2D6-01  TCGA-FI-A2EU-01  \
Hugo_Symbol                                                      
A1BG               25.181230       262.848180       313.167401   
A1CF                0.065503         0.065503         1.965087   
A2BP1              -0.330564        -0.330564         2.612215   
A2LD1              78.794176        96.621951        76.935429   
A2M              5359.061666      3323.598942      3076.094518   

             TCGA-FI-A2EW-01  TCGA-FI-A2EX-01  TCGA-FI-A2F4-01  \
Hugo_Symbol                                                      
A1BG              220.477469       125.334264        33.620333   
A1CF                0.065503         0.065503         0.065503   
A2BP1               6.873259         0.313398        -0.330564   
A2LD1              89.385709       116.696788       117.643280   
A2M              5029.960408      3565.729631      4701.782363   

             TCGA-FI-A2F8-01  TCGA-FI-A2F9-01  
Hugo_Symbol                                    
A1BG              130.850770       171.376680  
A1CF                0.065503         0.065503  
A2BP1               2.533070         0.511109  
A2LD1              87.169002       131.744185  
A2M             10955.857094      8940.781139  

[5 rows x 527 columns]

In [9]:
data_ucec_hugo_mv = data_ucec_hugo.dropna()
data_ucec_hugo_mv.shape 

(17494, 527)

In [10]:
data_ucec_hugo_mv.head()

TCGA-2E-A9G8-01  TCGA-4E-A92E-01  TCGA-5B-A90C-01  \
Hugo_Symbol                                                      
A1BG                 24.2393          19.8087           9.0662   
A1CF                  0.5157           0.0000           0.3163   
A2BP1                 0.0000           1.3661           1.5817   
A2LD1                73.8834         164.3720          75.2467   
A2M                6611.9100       13119.3000        3898.1000   

             TCGA-5S-A9Q8-01  TCGA-A5-A1OH-01  TCGA-A5-A2K2-01  \
Hugo_Symbol                                                      
A1BG                 60.9811         251.1470         128.9960   
A1CF                  0.0000           0.0000           0.0000   
A2BP1                 0.0000           2.2263           0.0000   
A2LD1                71.9793         239.0980          97.9912   
A2M                4464.3200        7509.7600        3453.6900   

             TCGA-A5-A2K3-01  TCGA-A5-A2K4-01  TCGA-A5-A2K5-01  \
Hugo_Symbol                                                      
A1BG                470.8330          66.7799          151.617   
A1CF                  0.9251           0.0000            0.000   
A2BP1                 0.0000           0.5394            0.000   
A2LD1               165.7820         109.0610          185.920   
A2M                8335.5600        7725.9900         5675.450   

             TCGA-A5-A2K7-01  ...  TCGA-FI-A2D2-01  TCGA-FI-A2D4-01  \
Hugo_Symbol                   ...                                     
A1BG                 99.4890  ...        33.660261        51.353706   
A1CF                  0.0000  ...         0.065503         0.065503   
A2BP1                 0.4667  ...        -0.330564        -0.330564   
A2LD1               154.0310  ...        57.648992       148.270803   
A2M                3424.5600  ...      6569.434216      4646.023218   

             TCGA-FI-A2D5-01  TCGA-FI-A2D6-01  TCGA-FI-A2EU-01  \
Hugo_Symbol                                                      
A1BG               25.181230       262.848180       313.167401   
A1CF                0.065503         0.065503         1.965087   
A2BP1              -0.330564        -0.330564         2.612215   
A2LD1              78.794176        96.621951        76.935429   
A2M              5359.061666      3323.598942      3076.094518   

             TCGA-FI-A2EW-01  TCGA-FI-A2EX-01  TCGA-FI-A2F4-01  \
Hugo_Symbol                                                      
A1BG              220.477469       125.334264        33.620333   
A1CF                0.065503         0.065503         0.065503   
A2BP1               6.873259         0.313398        -0.330564   
A2LD1              89.385709       116.696788       117.643280   
A2M              5029.960408      3565.729631      4701.782363   

             TCGA-FI-A2F8-01  TCGA-FI-A2F9-01  
Hugo_Symbol                                    
A1BG              130.850770       171.376680  
A1CF                0.065503         0.065503  
A2BP1               2.533070         0.511109  
A2LD1              87.169002       131.744185  
A2M             10955.857094      8940.781139  

[5 rows x 527 columns]

Após a aplicação da função 'dropna', foi possível eliminar os missing values (NaN) ainda existentes no dataset, uma vez que esta função elimina as linhas que apresentem pelo menos um missing value. Daí termos passado de 20502 linhas para 17494.

#### Retirar Duplicados

In [11]:
new_data = data_ucec_hugo_mv.reset_index()
new_data.drop_duplicates(subset=['Hugo_Symbol'], keep = False)

data = new_data.set_index("Hugo_Symbol")
data.shape

(17494, 527)

Observando os valores obtidos verifica-se que não existem duplicados, uma vez que após esta etapa de pré-processamento o número de linhas é igual ao obtido na etapa anterior 

### Metadados

In [12]:
data_patient = pd.read_csv('/Users/adrianacosta/Documents/GitHub/Bioinformatica_23-24/Dataset4/data_clinical_patient.txt', sep='\t', index_col = 0)
data_sample = pd.read_csv('/Users/adrianacosta/Documents/GitHub/Bioinformatica_23-24/Dataset4/data_clinical_sample.txt', sep='\t', index_col = 0)

#### data_patient

In [13]:
data_patient.head()

Subtype  \
#Patient Identifier                                        
#Identifier to uniquely specify a patient.       Subtype   
#STRING                                           STRING   
#1                                                     1   
PATIENT_ID                                       SUBTYPE   
TCGA-2E-A9G8                                UCEC_CN_HIGH   

                                                         TCGA PanCanAtlas Cancer Type Acronym  \
#Patient Identifier                                                                             
#Identifier to uniquely specify a patient.  Text field to hold cancer type acronym used by...   
#STRING                                                                                STRING   
#1                                                                                          1   
PATIENT_ID                                                                CANCER_TYPE_ACRONYM   
TCGA-2E-A9G8                                                                             UCEC   

                                                                   Other Patient ID  \
#Patient Identifier                                                                   
#Identifier to uniquely specify a patient.  Legacy DMP patient identifier (DMPnnnn)   
#STRING                                                                      STRING   
#1                                                                                1   
PATIENT_ID                                                         OTHER_PATIENT_ID   
TCGA-2E-A9G8                                   9583C10B-B21A-4863-98FA-61E735E64EA5   

                                                                                Diagnosis Age  \
#Patient Identifier                                                                             
#Identifier to uniquely specify a patient.  Age at which a condition or disease was first ...   
#STRING                                                                                NUMBER   
#1                                                                                          1   
PATIENT_ID                                                                                AGE   
TCGA-2E-A9G8                                                                               59   

                                               Sex  \
#Patient Identifier                                  
#Identifier to uniquely specify a patient.     Sex   
#STRING                                     STRING   
#1                                               1   
PATIENT_ID                                     SEX   
TCGA-2E-A9G8                                Female   

                                           Neoplasm Disease Stage American Joint Committee on Cancer Code  \
#Patient Identifier                                                                                         
#Identifier to uniquely specify a patient.  The extent of a cancer, especially whether the...               
#STRING                                                                                STRING               
#1                                                                                          1               
PATIENT_ID                                                        AJCC_PATHOLOGIC_TUMOR_STAGE               
TCGA-2E-A9G8                                                                              NaN               

                                           American Joint Committee on Cancer Publication Version Type  \
#Patient Identifier                                                                                      
#Identifier to uniquely specify a patient.  The version or edition of the American Joint C...            
#STRING                                                                                STRING            
#1                                                                                         

In [14]:
data_patient.columns = data_patient.iloc[3].values
data_patient = data_patient[4:]
data_patient.head()

SUBTYPE CANCER_TYPE_ACRONYM  \
#Patient Identifier                                     
TCGA-2E-A9G8         UCEC_CN_HIGH                UCEC   
TCGA-4E-A92E          UCEC_CN_LOW                UCEC   
TCGA-5B-A90C         UCEC_CN_HIGH                UCEC   
TCGA-5S-A9Q8          UCEC_CN_LOW                UCEC   
TCGA-A5-A0G1            UCEC_POLE                UCEC   

                                         OTHER_PATIENT_ID AGE     SEX  \
#Patient Identifier                                                     
TCGA-2E-A9G8         9583C10B-B21A-4863-98FA-61E735E64EA5  59  Female   
TCGA-4E-A92E         B43DE98D-4BB1-41A1-91F2-CE4E7EA4D0CA  54  Female   
TCGA-5B-A90C         16AC4341-CF8F-45E2-B90B-2D12D5F74A59  69  Female   
TCGA-5S-A9Q8         8751429B-4A11-451E-B978-DC9E9DB0EB36  51  Female   
TCGA-A5-A0G1         53707bb3-426a-43cb-830f-3eeed930295f  67  Female   

                    AJCC_PATHOLOGIC_TUMOR_STAGE AJCC_STAGING_EDITION  \
#Patient Identifier                                                    
TCGA-2E-A9G8                                NaN                 2009   
TCGA-4E-A92E                                NaN                 2009   
TCGA-5B-A90C                                NaN                 2009   
TCGA-5S-A9Q8                                NaN                 2009   
TCGA-A5-A0G1                                NaN                  NaN   

                    DAYS_LAST_FOLLOWUP DAYS_TO_BIRTH  \
#Patient Identifier                                    
TCGA-2E-A9G8                      1249           NaN   
TCGA-4E-A92E                        13        -19818   
TCGA-5B-A90C                      1202        -25518   
TCGA-5S-A9Q8                       313        -18785   
TCGA-A5-A0G1                       NaN        -24477   

                    DAYS_TO_INITIAL_PATHOLOGIC_DIAGNOSIS  ...  \
#Patient Identifier                                       ...   
TCGA-2E-A9G8                                           0  ...   
TCGA-4E-A92E                                           0  ...   
TCGA-5B-A90C                                           0  ...   
TCGA-5S-A9Q8                                           0  ...   
TCGA-A5-A0G1                                           0  ...   

                                                  CENTER  \
#Patient Identifier                                        
TCGA-2E-A9G8         University of Kansas Medical Center   
TCGA-4E-A92E                          Molecular Response   
TCGA-5B-A90C                Medical College of Wisconsin   
TCGA-5S-A9Q8                                  Holy Cross   
TCGA-A5-A0G1                                Cedars Sinai   

                    IN_PANCANPATHWAYS_FREEZE OS_STATUS     OS_MONTHS  \
#Patient Identifier                                                    
TCGA-2E-A9G8                             Yes    LIVING  41.062563698   
TCGA-4E-A92E                             Yes    LIVING  0.4273925765   
TCGA-5B-A90C                             Yes    LIVING  39.517375152   
TCGA-5S-A9Q8                             Yes    LIVING  10.290298189   
TCGA-A5-A0G1                             Yes  DECEASED  106.88102048   

                                   DSS_STATUS    DSS_MONTHS   DFS_STATUS  \
#Patient Identifier                                                        
TCGA-2E-A9G8         ALIVE OR DEAD TUMOR FREE  41.062563698          NaN   
TCGA-4E-A92E         ALIVE OR DEAD TUMOR FREE  0.4273925765  DiseaseFree   
TCGA-5B-A90C         ALIVE OR DEAD TUMOR FREE  39.517375152          NaN   
TCGA-5S-A9Q8         ALIVE OR DEAD TUMOR FREE  10.290298189  DiseaseFree   
TCGA-A5-A0G1         ALIVE OR DEAD TUMOR FREE  106.88102048  DiseaseFree   

                       DFS_MONTHS   PFS_STATUS    PFS_MONTHS  
#Patient Identifier                                           
TCGA-2E-A9G8                  NaN  PROGRESSION  11.013577933  
TCGA-4E-A92E         0.4273925765     CENSORED  0.4273925765  
TCGA-5B-A90C                  NaN  PROGRESSION  1

In [15]:
data_patient.shape

(529, 37)

In [16]:
data_patient.describe() #Gera estatísticas descritivas do dataframe

SUBTYPE CANCER_TYPE_ACRONYM  \
count            507                 529   
unique             4                   1   
top     UCEC_CN_HIGH                UCEC   
freq             163                 529   

                            OTHER_PATIENT_ID  AGE     SEX  \
count                                    529  526     529   
unique                                   529   59       1   
top     9583C10B-B21A-4863-98FA-61E735E64EA5   60  Female   
freq                                       1   28     529   

       AJCC_PATHOLOGIC_TUMOR_STAGE AJCC_STAGING_EDITION DAYS_LAST_FOLLOWUP  \
count                            0                  456                487   
unique                           0                    3                443   
top                            NaN                 2009                 39   
freq                           NaN                  277                  3   

       DAYS_TO_BIRTH DAYS_TO_INITIAL_PATHOLOGIC_DIAGNOSIS  ...       CENTER  \
count            525                                  523  ...          529   
unique           519                                    1  ...           29   
top           -22200                                    0  ...  Mayo Clinic   
freq               2                                  523  ...           76   

       IN_PANCANPATHWAYS_FREEZE OS_STATUS     OS_MONTHS  \
count                       529       529           528   
unique                        2         2           475   
top                         Yes    LIVING  41.062563698   
freq                        507       442             3   

                      DSS_STATUS    DSS_MONTHS   DFS_STATUS    DFS_MONTHS  \
count                        527           528          414           414   
unique                         2           475            2           380   
top     ALIVE OR DEAD TUMOR FREE  41.062563698  DiseaseFree  16.865568597   
freq                         469             3          359             3   

       PFS_STATUS    PFS_MONTHS  
count         529           528  
unique          2           466  
top      CENSORED  16.964197653  
freq          410             3  

[4 rows x 37 columns]

A contagem refere-se ao número de valores não nulos numa determinada coluna do dataset, por exemplo, na coluna Subtype existem 511 valores não nulos.

O parâmetro unique corresponde aos valores únicos, ou seja, não existe nenhum outro valor igual a si. 

O parâmetro top é referente à moda, ou seja, o atributo mais comum de uma determinada coluna.

A frequência diz-nos quantas vezes o número de contagens da moda, ou seja, das 511 contagens a moda aparece 163 vezes.

### Retirar missing values

In [17]:
data_patient = data_patient.dropna(axis = 1, how = 'all') 

In [18]:
data_patient.shape

(529, 32)

Nesta etapa, procedeu-se à eliminação das colunas que apresentavam todos os valores como missing values (NaN). Verificou-se que foram eliminadas 5 colunas.

In [19]:
del data_patient['DAYS_TO_INITIAL_PATHOLOGIC_DIAGNOSIS']

In [20]:
data_patient.head()

SUBTYPE CANCER_TYPE_ACRONYM  \
#Patient Identifier                                     
TCGA-2E-A9G8         UCEC_CN_HIGH                UCEC   
TCGA-4E-A92E          UCEC_CN_LOW                UCEC   
TCGA-5B-A90C         UCEC_CN_HIGH                UCEC   
TCGA-5S-A9Q8          UCEC_CN_LOW                UCEC   
TCGA-A5-A0G1            UCEC_POLE                UCEC   

                                         OTHER_PATIENT_ID AGE     SEX  \
#Patient Identifier                                                     
TCGA-2E-A9G8         9583C10B-B21A-4863-98FA-61E735E64EA5  59  Female   
TCGA-4E-A92E         B43DE98D-4BB1-41A1-91F2-CE4E7EA4D0CA  54  Female   
TCGA-5B-A90C         16AC4341-CF8F-45E2-B90B-2D12D5F74A59  69  Female   
TCGA-5S-A9Q8         8751429B-4A11-451E-B978-DC9E9DB0EB36  51  Female   
TCGA-A5-A0G1         53707bb3-426a-43cb-830f-3eeed930295f  67  Female   

                    AJCC_STAGING_EDITION DAYS_LAST_FOLLOWUP DAYS_TO_BIRTH  \
#Patient Identifier                                                         
TCGA-2E-A9G8                        2009               1249           NaN   
TCGA-4E-A92E                        2009                 13        -19818   
TCGA-5B-A90C                        2009               1202        -25518   
TCGA-5S-A9Q8                        2009                313        -18785   
TCGA-A5-A0G1                         NaN                NaN        -24477   

                                  ETHNICITY FORM_COMPLETION_DATE  ...  \
#Patient Identifier                                               ...   
TCGA-2E-A9G8         Not Hispanic Or Latino              5/21/14  ...   
TCGA-4E-A92E         Not Hispanic Or Latino              8/20/14  ...   
TCGA-5B-A90C         Not Hispanic Or Latino              4/17/14  ...   
TCGA-5S-A9Q8         Not Hispanic Or Latino             12/23/13  ...   
TCGA-A5-A0G1         Not Hispanic Or Latino               1/7/11  ...   

                                                  CENTER  \
#Patient Identifier                                        
TCGA-2E-A9G8         University of Kansas Medical Center   
TCGA-4E-A92E                          Molecular Response   
TCGA-5B-A90C                Medical College of Wisconsin   
TCGA-5S-A9Q8                                  Holy Cross   
TCGA-A5-A0G1                                Cedars Sinai   

                    IN_PANCANPATHWAYS_FREEZE OS_STATUS     OS_MONTHS  \
#Patient Identifier                                                    
TCGA-2E-A9G8                             Yes    LIVING  41.062563698   
TCGA-4E-A92E                             Yes    LIVING  0.4273925765   
TCGA-5B-A90C                             Yes    LIVING  39.517375152   
TCGA-5S-A9Q8                             Yes    LIVING  10.290298189   
TCGA-A5-A0G1                             Yes  DECEASED  106.88102048   

                                   DSS_STATUS    DSS_MONTHS   DFS_STATUS  \
#Patient Identifier                                                        
TCGA-2E-A9G8         ALIVE OR DEAD TUMOR FREE  41.062563698          NaN   
TCGA-4E-A92E         ALIVE OR DEAD TUMOR FREE  0.4273925765  DiseaseFree   
TCGA-5B-A90C         ALIVE OR DEAD TUMOR FREE  39.517375152          NaN   
TCGA-5S-A9Q8         ALIVE OR DEAD TUMOR FREE  10.290298189  DiseaseFree   
TCGA-A5-A0G1         ALIVE OR DEAD TUMOR FREE  106.88102048  DiseaseFree   

                       DFS_MONTHS   PFS_STATUS    PFS_MONTHS  
#Patient Identifier                                           
TCGA-2E-A9G8                  NaN  PROGRESSION  11.013577933  
TCGA-4E-A92E         0.4273925765     CENSORED  0.4273925765  
TCGA-5B-A90C                  NaN  PROGRESSION  16.997074005  
TCGA-5S-A9Q8         10.290298189     CENSORED  10.290298189  
TCGA-A5-A0G1         106.88102048     CENSORED  106.88102048  

[5 rows x 31 columns]

In [21]:
data_patient.shape

(529, 31)

Procedeu-se à eliminação da coluna 'DAYS_TO_INITIAL_PATHOLOGIC_DIAGNOSIS', uma vez que esta apresentava todos os seus atributos a 0. Para isso, utilizou-se a função "del". 

### Retirar colunas repetidas

In [22]:
data_patient['OS_MONTHS'].equals(data_patient['DSS_MONTHS']) 
if data_patient['OS_MONTHS'].equals(data_patient['DSS_MONTHS']) == True:
  del data_patient['DSS_MONTHS']

In [23]:
data_patient.head()

SUBTYPE CANCER_TYPE_ACRONYM  \
#Patient Identifier                                     
TCGA-2E-A9G8         UCEC_CN_HIGH                UCEC   
TCGA-4E-A92E          UCEC_CN_LOW                UCEC   
TCGA-5B-A90C         UCEC_CN_HIGH                UCEC   
TCGA-5S-A9Q8          UCEC_CN_LOW                UCEC   
TCGA-A5-A0G1            UCEC_POLE                UCEC   

                                         OTHER_PATIENT_ID AGE     SEX  \
#Patient Identifier                                                     
TCGA-2E-A9G8         9583C10B-B21A-4863-98FA-61E735E64EA5  59  Female   
TCGA-4E-A92E         B43DE98D-4BB1-41A1-91F2-CE4E7EA4D0CA  54  Female   
TCGA-5B-A90C         16AC4341-CF8F-45E2-B90B-2D12D5F74A59  69  Female   
TCGA-5S-A9Q8         8751429B-4A11-451E-B978-DC9E9DB0EB36  51  Female   
TCGA-A5-A0G1         53707bb3-426a-43cb-830f-3eeed930295f  67  Female   

                    AJCC_STAGING_EDITION DAYS_LAST_FOLLOWUP DAYS_TO_BIRTH  \
#Patient Identifier                                                         
TCGA-2E-A9G8                        2009               1249           NaN   
TCGA-4E-A92E                        2009                 13        -19818   
TCGA-5B-A90C                        2009               1202        -25518   
TCGA-5S-A9Q8                        2009                313        -18785   
TCGA-A5-A0G1                         NaN                NaN        -24477   

                                  ETHNICITY FORM_COMPLETION_DATE  ... WEIGHT  \
#Patient Identifier                                               ...          
TCGA-2E-A9G8         Not Hispanic Or Latino              5/21/14  ...     71   
TCGA-4E-A92E         Not Hispanic Or Latino              8/20/14  ...     76   
TCGA-5B-A90C         Not Hispanic Or Latino              4/17/14  ...     98   
TCGA-5S-A9Q8         Not Hispanic Or Latino             12/23/13  ...     86   
TCGA-A5-A0G1         Not Hispanic Or Latino               1/7/11  ...     59   

                                                  CENTER  \
#Patient Identifier                                        
TCGA-2E-A9G8         University of Kansas Medical Center   
TCGA-4E-A92E                          Molecular Response   
TCGA-5B-A90C                Medical College of Wisconsin   
TCGA-5S-A9Q8                                  Holy Cross   
TCGA-A5-A0G1                                Cedars Sinai   

                    IN_PANCANPATHWAYS_FREEZE OS_STATUS     OS_MONTHS  \
#Patient Identifier                                                    
TCGA-2E-A9G8                             Yes    LIVING  41.062563698   
TCGA-4E-A92E                             Yes    LIVING  0.4273925765   
TCGA-5B-A90C                             Yes    LIVING  39.517375152   
TCGA-5S-A9Q8                             Yes    LIVING  10.290298189   
TCGA-A5-A0G1                             Yes  DECEASED  106.88102048   

                                   DSS_STATUS   DFS_STATUS    DFS_MONTHS  \
#Patient Identifier                                                        
TCGA-2E-A9G8         ALIVE OR DEAD TUMOR FREE          NaN           NaN   
TCGA-4E-A92E         ALIVE OR DEAD TUMOR FREE  DiseaseFree  0.4273925765   
TCGA-5B-A90C         ALIVE OR DEAD TUMOR FREE          NaN           NaN   
TCGA-5S-A9Q8         ALIVE OR DEAD TUMOR FREE  DiseaseFree  10.290298189   
TCGA-A5-A0G1         ALIVE OR DEAD TUMOR FREE  DiseaseFree  106.88102048   

                      PFS_STATUS    PFS_MONTHS  
#Patient Identifier                             
TCGA-2E-A9G8         PROGRESSION  11.013577933  
TCGA-4E-A92E            CENSORED  0.4273925765  
TCGA-5B-A90C         PROGRESSION  16.997074005  
TCGA-5S-A9Q8            CENSORED  10.290298189  
TCGA-A5-A0G1            CENSORED  106.88102048  

[5 rows x 30 columns]

In [24]:
data_patient.shape

(529, 30)

Pela análise do dataset, verifiou-se que as colunas 'OS_MONTHS' e 'DSS_MONTHS' eram iguais. Desta forma, igualou-se e caso se verifique que estas são iguais elimina-se uma delas, neste caso a 'DSS_MONTHS'. Isto verifica-se pelo número de colunas resultantes.

### Retirar colunas com valores únicos

In [25]:
for col in data_patient.columns:
    if len(data_patient[col].unique()) == 1:
        data_patient.drop(col,inplace = True, axis = 1)

In [26]:
data_patient.shape

(529, 27)

Procedeu-se à eliminação das colunas que possuíam valores únicos, isto é, todos os valores iguais. Esta etapa é importante, pois estas colunas não apresentam informação relevante para o estudo, uma vez que não possuem variações para os diferentes pacientes. 

#### data_sample

In [27]:
data_sample.head()

Sample Identifier  \
#Patient Identifier                                                       
#Identifier to uniquely specify a patient.  A unique sample identifier.   
#STRING                                                          STRING   
#1                                                                    1   
PATIENT_ID                                                    SAMPLE_ID   
TCGA-2E-A9G8                                            TCGA-2E-A9G8-01   

                                            Oncotree Code  \
#Patient Identifier                                         
#Identifier to uniquely specify a patient.  Oncotree Code   
#STRING                                            STRING   
#1                                                      1   
PATIENT_ID                                  ONCOTREE_CODE   
TCGA-2E-A9G8                                          UEC   

                                                      Cancer Type  \
#Patient Identifier                                                 
#Identifier to uniquely specify a patient.            Cancer Type   
#STRING                                                    STRING   
#1                                                              1   
PATIENT_ID                                            CANCER_TYPE   
TCGA-2E-A9G8                                Endometrial Carcinoma   

                                                      Cancer Type Detailed  \
#Patient Identifier                                                          
#Identifier to uniquely specify a patient.            Cancer Type Detailed   
#STRING                                                             STRING   
#1                                                                       1   
PATIENT_ID                                            CANCER_TYPE_DETAILED   
TCGA-2E-A9G8                                Uterine Endometrioid Carcinoma   

                                                                         Tumor Type  \
#Patient Identifier                                                                   
#Identifier to uniquely specify a patient.                               Tumor Type   
#STRING                                                                      STRING   
#1                                                                                1   
PATIENT_ID                                                               TUMOR_TYPE   
TCGA-2E-A9G8                                Endometrioid Endometrial Adenocarcinoma   

                                                                    Neoplasm Histologic Grade  \
#Patient Identifier                                                                             
#Identifier to uniquely specify a patient.  Numeric value to express the degree of abnorma...   
#STRING                                                                                STRING   
#1                                                                                          1   
PATIENT_ID                                                                              GRADE   
TCGA-2E-A9G8                                                                               G2   

                                             Tissue Prospective Collection Indicator  \
#Patient Identifier                                                                    
#Identifier to uniquely specify a patient.  Tissue prospective collection indicator.   
#STRING                                                                       STRING   
#1                                                                                 1   
PATIENT_ID                                   TISSUE_PROSPECTIVE_COLLECTION_INDICATOR   
TCGA-2E-A9G8                                                                      No   

                                             Tissue Retrospective Collection Indicator  \
#Patient Identifier                                                     

In [28]:
data_sample.columns = data_sample.iloc[3].values
data_sample = data_sample[4:]

data_sample.head()

SAMPLE_ID ONCOTREE_CODE            CANCER_TYPE  \
#Patient Identifier                                                         
TCGA-2E-A9G8         TCGA-2E-A9G8-01           UEC  Endometrial Carcinoma   
TCGA-4E-A92E         TCGA-4E-A92E-01           UEC  Endometrial Carcinoma   
TCGA-5B-A90C         TCGA-5B-A90C-01           UEC  Endometrial Carcinoma   
TCGA-5S-A9Q8         TCGA-5S-A9Q8-01           UEC  Endometrial Carcinoma   
TCGA-A5-A0G1         TCGA-A5-A0G1-01           USC  Endometrial Carcinoma   

                                                  CANCER_TYPE_DETAILED  \
#Patient Identifier                                                      
TCGA-2E-A9G8                            Uterine Endometrioid Carcinoma   
TCGA-4E-A92E                            Uterine Endometrioid Carcinoma   
TCGA-5B-A90C                            Uterine Endometrioid Carcinoma   
TCGA-5S-A9Q8                            Uterine Endometrioid Carcinoma   
TCGA-A5-A0G1         Uterine Serous Carcinoma/Uterine Papillary Ser...   

                                                  TUMOR_TYPE GRADE  \
#Patient Identifier                                                  
TCGA-2E-A9G8         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-4E-A92E         Endometrioid Endometrial Adenocarcinoma    G1   
TCGA-5B-A90C         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-5S-A9Q8         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-A5-A0G1               Serous Endometrial Adenocarcinoma    G3   

                    TISSUE_PROSPECTIVE_COLLECTION_INDICATOR  \
#Patient Identifier                                           
TCGA-2E-A9G8                                             No   
TCGA-4E-A92E                                            Yes   
TCGA-5B-A90C                                             No   
TCGA-5S-A9Q8                                            Yes   
TCGA-A5-A0G1                                             No   

                    TISSUE_RETROSPECTIVE_COLLECTION_INDICATOR  \
#Patient Identifier                                             
TCGA-2E-A9G8                                              Yes   
TCGA-4E-A92E                                               No   
TCGA-5B-A90C                                              Yes   
TCGA-5S-A9Q8                                               No   
TCGA-A5-A0G1                                              Yes   

                    TISSUE_SOURCE_SITE TUMOR_TISSUE_SITE  ...  STATUS_17Q  \
#Patient Identifier                                       ...               
TCGA-2E-A9G8                        2E            Uterus  ...         NaN   
TCGA-4E-A92E                        4E            Uterus  ...  Not Called   
TCGA-5B-A90C                        5B            Uterus  ...         NaN   
TCGA-5S-A9Q8                        5S            Uterus  ...  Not Called   
TCGA-A5-A0G1                        A5            Uterus  ...         NaN   

                     STATUS_18P  STATUS_18Q  STATUS_19P  STATUS_19Q  \
#Patient Identifier                                                   
TCGA-2E-A9G8                NaN         NaN         NaN         NaN   
TCGA-4E-A92E         Not Called  Not Called  Not Called  Not Called   
TCGA-5B-A90C             Gained      Gained         NaN         NaN   
TCGA-5S-A9Q8         Not Called  Not Called  Not Called  Not Called   
TCGA-A5-A0G1                NaN         NaN         NaN         NaN   

                     STATUS_20P  STATUS_20Q STATUS_21_21Q STATUS_22_22Q  \
#Patient Identifier                                                       
TCGA-2E-A9G8               Lost      Gained    Not Called          Lost   
TCGA-4E-A92E         Not Called  Not Called    Not Called    Not Called   
TCGA-5B-A90C         Not Called      Gained    Not Called    Not Called   
TCGA-5S-A9Q8         Not Called  Not Called    Not Called    Not Called   
TCGA-A5-A0G1                NaN         NaN           NaN           NaN   

               

In [29]:
data_sample.shape

(529, 51)

In [30]:
data_sample.describe() #Gera estatísticas descritivas do dataframe

SAMPLE_ID ONCOTREE_CODE            CANCER_TYPE  \
count               529           529                    529   
unique              529             3                      1   
top     TCGA-2E-A9G8-01           UEC  Endometrial Carcinoma   
freq                  1           399                    529   

                  CANCER_TYPE_DETAILED  \
count                              529   
unique                               3   
top     Uterine Endometrioid Carcinoma   
freq                               399   

                                     TUMOR_TYPE GRADE  \
count                                       529   529   
unique                                        3     4   
top     Endometrioid Endometrial Adenocarcinoma    G3   
freq                                        399   302   

       TISSUE_PROSPECTIVE_COLLECTION_INDICATOR  \
count                                      524   
unique                                       2   
top                                         No   
freq                                       416   

       TISSUE_RETROSPECTIVE_COLLECTION_INDICATOR TISSUE_SOURCE_SITE  \
count                                        529                529   
unique                                         2                 30   
top                                          Yes                 D1   
freq                                         421                 76   

       TUMOR_TISSUE_SITE  ...  STATUS_17Q  STATUS_18P  STATUS_18Q  STATUS_19P  \
count                529  ...         445         497         476         408   
unique                 2  ...           3           3           3           3   
top               Uterus  ...  Not Called  Not Called  Not Called  Not Called   
freq                 528  ...         376         411         399         368   

        STATUS_19Q  STATUS_20P  STATUS_20Q STATUS_21_21Q STATUS_22_22Q  \
count          454         492         481           493           467   
unique           3           3           3             3             3   
top     Not Called  Not Called  Not Called    Not Called    Not Called   
freq           405         418         396           426           372   

       SAMPLE_TYPE  
count          529  
unique           1  
top        Primary  
freq           529  

[4 rows x 51 columns]

No parâmetro top aparece 'A unique sample identifier' porque cada id é único, e portanto não existe uma moda.

Na sample_id a frequência deu 1 porque todas as amostras vão ter um id diferente, enquanto que nas outras categorias, como a frequência é diferente de 1, quer dizer que não têm uma resposta única.

### Retirar missing values

In [31]:
data_sample = data_sample.dropna(axis = 1, how = 'all') 

In [32]:
data_sample.head()

SAMPLE_ID ONCOTREE_CODE            CANCER_TYPE  \
#Patient Identifier                                                         
TCGA-2E-A9G8         TCGA-2E-A9G8-01           UEC  Endometrial Carcinoma   
TCGA-4E-A92E         TCGA-4E-A92E-01           UEC  Endometrial Carcinoma   
TCGA-5B-A90C         TCGA-5B-A90C-01           UEC  Endometrial Carcinoma   
TCGA-5S-A9Q8         TCGA-5S-A9Q8-01           UEC  Endometrial Carcinoma   
TCGA-A5-A0G1         TCGA-A5-A0G1-01           USC  Endometrial Carcinoma   

                                                  CANCER_TYPE_DETAILED  \
#Patient Identifier                                                      
TCGA-2E-A9G8                            Uterine Endometrioid Carcinoma   
TCGA-4E-A92E                            Uterine Endometrioid Carcinoma   
TCGA-5B-A90C                            Uterine Endometrioid Carcinoma   
TCGA-5S-A9Q8                            Uterine Endometrioid Carcinoma   
TCGA-A5-A0G1         Uterine Serous Carcinoma/Uterine Papillary Ser...   

                                                  TUMOR_TYPE GRADE  \
#Patient Identifier                                                  
TCGA-2E-A9G8         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-4E-A92E         Endometrioid Endometrial Adenocarcinoma    G1   
TCGA-5B-A90C         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-5S-A9Q8         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-A5-A0G1               Serous Endometrial Adenocarcinoma    G3   

                    TISSUE_PROSPECTIVE_COLLECTION_INDICATOR  \
#Patient Identifier                                           
TCGA-2E-A9G8                                             No   
TCGA-4E-A92E                                            Yes   
TCGA-5B-A90C                                             No   
TCGA-5S-A9Q8                                            Yes   
TCGA-A5-A0G1                                             No   

                    TISSUE_RETROSPECTIVE_COLLECTION_INDICATOR  \
#Patient Identifier                                             
TCGA-2E-A9G8                                              Yes   
TCGA-4E-A92E                                               No   
TCGA-5B-A90C                                              Yes   
TCGA-5S-A9Q8                                               No   
TCGA-A5-A0G1                                              Yes   

                    TISSUE_SOURCE_SITE TUMOR_TISSUE_SITE  ...  STATUS_17Q  \
#Patient Identifier                                       ...               
TCGA-2E-A9G8                        2E            Uterus  ...         NaN   
TCGA-4E-A92E                        4E            Uterus  ...  Not Called   
TCGA-5B-A90C                        5B            Uterus  ...         NaN   
TCGA-5S-A9Q8                        5S            Uterus  ...  Not Called   
TCGA-A5-A0G1                        A5            Uterus  ...         NaN   

                     STATUS_18P  STATUS_18Q  STATUS_19P  STATUS_19Q  \
#Patient Identifier                                                   
TCGA-2E-A9G8                NaN         NaN         NaN         NaN   
TCGA-4E-A92E         Not Called  Not Called  Not Called  Not Called   
TCGA-5B-A90C             Gained      Gained         NaN         NaN   
TCGA-5S-A9Q8         Not Called  Not Called  Not Called  Not Called   
TCGA-A5-A0G1                NaN         NaN         NaN         NaN   

                     STATUS_20P  STATUS_20Q STATUS_21_21Q STATUS_22_22Q  \
#Patient Identifier                                                       
TCGA-2E-A9G8               Lost      Gained    Not Called          Lost   
TCGA-4E-A92E         Not Called  Not Called    Not Called    Not Called   
TCGA-5B-A90C         Not Called      Gained    Not Called    Not Called   
TCGA-5S-A9Q8         Not Called  Not Called    Not Called    Not Called   
TCGA-A5-A0G1                NaN         NaN           NaN           NaN   

               

In [33]:
data_sample.shape

(529, 51)

É possível concluir que não existe nenhuma coluna, neste dataset, em que todos os atributos são missing values.

### Retirar colunas com valores únicos

In [34]:
for col in data_sample.columns:
    if len(data_sample[col].unique()) == 1:
        data_sample.drop(col,inplace = True, axis = 1)

In [35]:
data_sample.shape

(529, 49)

Foram eliminadas 2 colunas, pois estas apresentavam apenas 1 valor único.

### Unir metadados

In [13]:
unified_meta = data_sample.join(data_patient)

In [14]:
unified_meta.shape

(529, 88)

In [15]:
unified_meta.head()

SAMPLE_ID ONCOTREE_CODE            CANCER_TYPE  \
#Patient Identifier                                                         
TCGA-2E-A9G8         TCGA-2E-A9G8-01           UEC  Endometrial Carcinoma   
TCGA-4E-A92E         TCGA-4E-A92E-01           UEC  Endometrial Carcinoma   
TCGA-5B-A90C         TCGA-5B-A90C-01           UEC  Endometrial Carcinoma   
TCGA-5S-A9Q8         TCGA-5S-A9Q8-01           UEC  Endometrial Carcinoma   
TCGA-A5-A0G1         TCGA-A5-A0G1-01           USC  Endometrial Carcinoma   

                                                  CANCER_TYPE_DETAILED  \
#Patient Identifier                                                      
TCGA-2E-A9G8                            Uterine Endometrioid Carcinoma   
TCGA-4E-A92E                            Uterine Endometrioid Carcinoma   
TCGA-5B-A90C                            Uterine Endometrioid Carcinoma   
TCGA-5S-A9Q8                            Uterine Endometrioid Carcinoma   
TCGA-A5-A0G1         Uterine Serous Carcinoma/Uterine Papillary Ser...   

                                                  TUMOR_TYPE GRADE  \
#Patient Identifier                                                  
TCGA-2E-A9G8         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-4E-A92E         Endometrioid Endometrial Adenocarcinoma    G1   
TCGA-5B-A90C         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-5S-A9Q8         Endometrioid Endometrial Adenocarcinoma    G2   
TCGA-A5-A0G1               Serous Endometrial Adenocarcinoma    G3   

                    TISSUE_PROSPECTIVE_COLLECTION_INDICATOR  \
#Patient Identifier                                           
TCGA-2E-A9G8                                             No   
TCGA-4E-A92E                                            Yes   
TCGA-5B-A90C                                             No   
TCGA-5S-A9Q8                                            Yes   
TCGA-A5-A0G1                                             No   

                    TISSUE_RETROSPECTIVE_COLLECTION_INDICATOR  \
#Patient Identifier                                             
TCGA-2E-A9G8                                              Yes   
TCGA-4E-A92E                                               No   
TCGA-5B-A90C                                              Yes   
TCGA-5S-A9Q8                                               No   
TCGA-A5-A0G1                                              Yes   

                    TISSUE_SOURCE_SITE TUMOR_TISSUE_SITE  ...  \
#Patient Identifier                                       ...   
TCGA-2E-A9G8                        2E            Uterus  ...   
TCGA-4E-A92E                        4E            Uterus  ...   
TCGA-5B-A90C                        5B            Uterus  ...   
TCGA-5S-A9Q8                        5S            Uterus  ...   
TCGA-A5-A0G1                        A5            Uterus  ...   

                                                  CENTER  \
#Patient Identifier                                        
TCGA-2E-A9G8         University of Kansas Medical Center   
TCGA-4E-A92E                          Molecular Response   
TCGA-5B-A90C                Medical College of Wisconsin   
TCGA-5S-A9Q8                                  Holy Cross   
TCGA-A5-A0G1                                Cedars Sinai   

                    IN_PANCANPATHWAYS_FREEZE OS_STATUS     OS_MONTHS  \
#Patient Identifier                                                    
TCGA-2E-A9G8                             Yes    LIVING  41.062563698   
TCGA-4E-A92E                             Yes    LIVING  0.4273925765   
TCGA-5B-A90C                             Yes    LIVING  39.517375152   
TCGA-5S-A9Q8                             Yes    LIVING  10.290298189   
TCGA-A5-A0G1                             Yes  DECEASED  106.88102048   

                                   DSS_STATUS    DSS_MONTHS   DFS_STATUS  \
#Patient Identifier                                                        
TCGA-2E-A9G8         ALIVE OR DEAD TUMOR FREE

In [16]:
unified_meta.describe()

SAMPLE_ID ONCOTREE_CODE            CANCER_TYPE  \
count               529           529                    529   
unique              529             3                      1   
top     TCGA-2E-A9G8-01           UEC  Endometrial Carcinoma   
freq                  1           399                    529   

                  CANCER_TYPE_DETAILED  \
count                              529   
unique                               3   
top     Uterine Endometrioid Carcinoma   
freq                               399   

                                     TUMOR_TYPE GRADE  \
count                                       529   529   
unique                                        3     4   
top     Endometrioid Endometrial Adenocarcinoma    G3   
freq                                        399   302   

       TISSUE_PROSPECTIVE_COLLECTION_INDICATOR  \
count                                      524   
unique                                       2   
top                                         No   
freq                                       416   

       TISSUE_RETROSPECTIVE_COLLECTION_INDICATOR TISSUE_SOURCE_SITE  \
count                                        529                529   
unique                                         2                 30   
top                                          Yes                 D1   
freq                                         421                 76   

       TUMOR_TISSUE_SITE  ...       CENTER IN_PANCANPATHWAYS_FREEZE OS_STATUS  \
count                529  ...          529                      529       529   
unique                 2  ...           29                        2         2   
top               Uterus  ...  Mayo Clinic                      Yes    LIVING   
freq                 528  ...           76                      507       442   

           OS_MONTHS                DSS_STATUS    DSS_MONTHS   DFS_STATUS  \
count            528                       527           528          414   
unique           475                         2           475            2   
top     41.062563698  ALIVE OR DEAD TUMOR FREE  41.062563698  DiseaseFree   
freq               3                       469             3          359   

          DFS_MONTHS PFS_STATUS    PFS_MONTHS  
count            414        529           528  
unique           380          2           466  
top     16.865568597   CENSORED  16.964197653  
freq               3        410             3  

[4 rows x 88 columns]

In [17]:
unified_meta = unified_meta.set_index("SAMPLE_ID")

In [18]:
unified_meta = unified_meta.loc[data_ucec_hugo.columns,:]

In [19]:
unified_meta.describe()

ONCOTREE_CODE            CANCER_TYPE            CANCER_TYPE_DETAILED  \
count            527                    527                             527   
unique             3                      1                               3   
top              UEC  Endometrial Carcinoma  Uterine Endometrioid Carcinoma   
freq             397                    527                             397   

                                     TUMOR_TYPE GRADE  \
count                                       527   527   
unique                                        3     4   
top     Endometrioid Endometrial Adenocarcinoma    G3   
freq                                        397   301   

       TISSUE_PROSPECTIVE_COLLECTION_INDICATOR  \
count                                      522   
unique                                       2   
top                                         No   
freq                                       415   

       TISSUE_RETROSPECTIVE_COLLECTION_INDICATOR TISSUE_SOURCE_SITE  \
count                                        527                527   
unique                                         2                 30   
top                                          Yes                 D1   
freq                                         420                 76   

       TUMOR_TISSUE_SITE ANEUPLOIDY_SCORE  ...       CENTER  \
count                527              514  ...          527   
unique                 2               30  ...           29   
top               Uterus                0  ...  Mayo Clinic   
freq                 526              147  ...           76   

       IN_PANCANPATHWAYS_FREEZE OS_STATUS     OS_MONTHS  \
count                       527       527           526   
unique                        2         2           473   
top                         Yes    LIVING  41.062563698   
freq                        507       440             3   

                      DSS_STATUS    DSS_MONTHS   DFS_STATUS    DFS_MONTHS  \
count                        525           526          412           412   
unique                         2           473            2           378   
top     ALIVE OR DEAD TUMOR FREE  41.062563698  DiseaseFree  16.865568597   
freq                         467             3          357             3   

       PFS_STATUS    PFS_MONTHS  
count         527           526  
unique          2           464  
top      CENSORED  16.865568597  
freq          408             3  

[4 rows x 87 columns]

In [20]:
unified_meta.shape

(527, 87)